packages/api/src/services/actions.ts

```diff
@@ -89,6 +89,26 @@ export function Handlers(config: Config, appState: Dependencies): Actions {
       // convert this to tuple to save bytes.
       return results.map(({ price, timestamp }) => [timestamp, price]);
     },
+    async historicalSynthPricesByAddress(
+      empAddress: string,
+      start = 0,
+      end: number = Date.now()
+    ): Promise<PriceSample[]> {
+      assert(empAddress, "requires emp address");
+      assert(start >= 0, "requires a start value >= 0");
+      assert(exists(synthPrices.history[empAddress]), "No synthetic prices for emp address: " + empAddress);
+      const results = await synthPrices.history[empAddress].betweenByTimestamp(start, end);
+      // convert this to tuple to save bytes.
+      return results.map(({ price, timestamp }) => [timestamp, price]);
+    },
+    async sliceHistoricalSynthPricesByAddress(empAddress: string, start = 0, length = 1): Promise<PriceSample[]> {
+      assert(empAddress, "requires emp address");
```

Comment:
What is this asserting? Isn't its existence guaranteed by the typechecker? Or are you specifically asserting that it's not an empty string?

Reply:
all the action calls can be incorrectly called since these are exposed by express and called by a third party, so more checks need to be added here to give the client feedback.

Reply:
I see, so this _is_ the API. That makes sense!

One question on that note: it seems like the caller needs to specify these parameters in the correct order in the body of their request? Would it be less error-prone to use named params and just pass the body object directly?

```ts 
async sliceHistoricalSynthPricesByAddress({ empAddress: string, start = 0, length = 1 }): Promise<PriceSample[]> {
```

We don't have to do this now, but I think it would be a good quality of life improvement for the API users. Thoughts?

Reply:
absolutely could, the main tradeoff with it is slightly more verbose calls on the client side and more data over the wire, but it may allow a more intuitive api.  in the past i have done the calls in the way you suggest, so i am open to that.